src/utils/networking.cpp

```diff
@@ -0,0 +1,24 @@
+#include "utils/networking.hpp"
+
+namespace precice {
+namespace utils {
+namespace networking {
+
+std::string loopbackInterfaceName()
+{
+#if defined(__linux__)
+  return "lo";
+#elif defined(__APPLE__)
+  return "lo0";
+#elif defined(__WIN32)
+  // Not required as we directly use the 127.0.0.1 under Windows
+  return "";
+#else
+#warning "Your target architecture does not define a loopback interface. Please consider reporting this upstream."
```

Comment:
"upstream" could be ambiguous here: is it preCICE or some dependency? What if we use the "preCICE developers" here?

```suggestion
#warning "Your target architecture does not define a loopback interface. Please consider reporting this to the preCICE developers."
```

Reply:
I don't think that upstream is ambiguous as this warning will only be displayed when compiling preCICE itself.
That said, upstream is not a mainstream term (no pun intended). Your version is understandable by a wider audience. :+1: